pkg/bgpv1/README.md

```diff
@@ -0,0 +1,311 @@
+# Cilium BGP Control Plane
+
+## Usage
+
+Currently a single flag in the `Cilium Agent` exists to turn on the `BGP Control Plane` feature set.
+
+```
+--enable-bgp-control-plane=true
+```
+
+When set to `true` the `BGP Control Plane` `Controllers` will be instantiated and will begin listening for `CiliumBGPPeeringPolicy` events.
+
+### CiliumBGPPeeringPolicy CRD
+
+All `BGP` peering topology information is carried in a `CiliumBGPPeeringPolicy` CRD. 
+
+`CiliumBGPPeeringPolicy` can be applied to one or more nodes based on its `nodeSelector` fields. 
+
+A Cilium node may only have a single `CiliumBGPPeeringPolicy` apply to it and if more then one does, it will apply no policy at all.
+
+Each `CiliumBGPPeeringPolicy` defines one or more `CiliumBGPVirtualRouter` configurations.
+
+When these CRDs are written or read from the cluster the `Controllers` will take notice and perform the necessary actions to drive the `BGP Control Plane` to the desired state described by the policy.
+
+The policy in `yaml` form is defined below:
+
+```yaml
+apiVersion: "cilium.io/v2alpha1"
+kind: CiliumBGPPeeringPolicy
+metadata:
+ name: 01-bgp-peering-policy
+spec: # CiliumBGPPeeringPolicySpec
+ nodeSelector:
+   matchLabels:
+     bgp-policy: a
+ virtualRouters: # []CiliumBGPVirtualRouter
+ - localASN: 64512
+   exportPodCIDR: true
+   neighbors: # []CiliumBGPNeighbor
+	 - peerAddress: 'fc00:f853:ccd:e793::50/128'
+	   peerASN: 64512
+```
+
+#### Fields
+``` 
+nodeSelector: Nodes which are selected by this label selector will apply the given policy
+ 
+ virtualRouters: One or more peering configurations outlined below. Each peering configuration can be thought of as a BGP router instance.
+	
+	virtualRouters[*].localASN: The local ASN for this peering configuration
+	
+	virtualRouters[*].exportPodCIDR: Whether to export the private pod CIDR block to the listed neighbors
+	
+	virtualRouters[*].neighbors: A list of neighbors to peer with
+		neighbors[*].peerAddress: The address of the peer neighbor
+		neighbors[*].peerASN: The ASN of the peer	
+```
+
+*note*: *setting unique configuration details of a particular instantiated virtual router on a particular Cilium node is explained in [Virtual Router Attributes](#Virtual Router Attributes)*
+
+## Creating  a BGP Topology
+
+### Rules
+
+The following rules must be adhered to for a `CiliumBGPPeeringPolicy` to correctly apply to a node.
+
+- Only a single `CiliumBGPPeeringPolicy` can apply to a `Cilium` node. 
+	* If the `BGP Control Plane` on a node iterates through the `CiliumBGPPeeringPolicy` CRs currently written to the cluster and discovers (n > 1) policies match its labels, it will return an error and remove any existing BGP sessions. Only (n == 1) policies **must** match a node's label sets.
+	* Administrators should test a new BGP topology in a staging environment before making permanent changes in production.
+
+```mermaid
```

Comment:
This one I don't quite understand. 

I think the mermaid diagram explains all three above points, they all deal with the situation where a policy will not apply.

Reply:
What I meant is that, as far as I remember, this:

```
* foo
  * bar

'''
some code, for example Mermaid
'''
explaining here that this diagram is awesome

* baz
```

... renders differently from:


```
* foo
  * bar
    ```
    some code, for example Mermaid
    ```
    explaining here that this diagram is awesome
* baz
```

The difference being that, in the first case, there are two distinct lists, with the code/diagram between them, whereas in the second case it is part of the list (part of one of the bullet items).

Given that you were providing a set of rules to follow, I thought that the remaining items below the diagrams were also part of this list, and that it would look better maybe to have a single list with the diagram part of it, instead of separate lists. But maybe I was wrong.

Reply:
Ah, I do think that hte diagram is a general explanation of the points above.